src/main/java/graphql/normalized/ExecutableNormalizedField.java

```diff
@@ -505,14 +502,69 @@ private void traverseImpl(ExecutableNormalizedField root,
         });
     }
 
+    /**
+     * This tries to find interfaces common to all the field output types.
+     * <p>
+     * i.e. goes through {@link #getFieldDefinitions(GraphQLSchema)} and finds an interface that
+     * all the field's unwrapped output types are assignable to.
+     */
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private Set<GraphQLInterfaceType> getInterfacesCommonToAllOutputTypes(GraphQLSchema schema) {
+        Ref<Set<GraphQLInterfaceType>> ref = new Ref<>();
```

Comment:
Why do you need this Ref class?

Set<GraphQLInterfaceType>> set = null;

if (set = null) {
  set = new LinkedHashSet()
} else {
  set.retainAll(x)
}

Reply:
It's Java. Can't touch non (effectively) final variables from inside lambdas.

(This gets used inside the `forEachFieldDefinitions` lambda).

It's basically a replacement for `AtomicReference` or `Set<GraphQLInterfaceType>[]` but more performant.

I will however move this down closer to the lambda usage.

Reply:
Ahh I get it - another trick is an array with one entry

```
Set<GraphQLInterfaceType> ref[] = {null}
```
and then use the one entry in a mutable way.

I like your approach.